kyaml/runfn/runfn.go

```diff
@@ -190,7 +190,8 @@ func (r RunFns) runFunctions(
 	} else {
 		// write to the output instead of the directory if r.Output is specified or
 		// the output is nil (reading from Input)
-		outputs = append(outputs, kio.ByteWriter{Writer: r.Output})
+		// remove PathAnnotation and LegacyPathAnnotation explicitly because ByteWriter won't clear them by default
+		outputs = append(outputs, kio.ByteWriter{Writer: r.Output, ClearAnnotations: []string{kioutil.PathAnnotation, kioutil.LegacyPathAnnotation}})
```

Comment:
We're adding this to the run function for the pathannotation and legacypathannotation, can we add more tests on other thing that may also be using these properties? Or is this quite isolated at the moment? Because I'm afraid that there may be another unexpected side effect that may happen.